### PR TITLE
FIX: Update the filename conventions and parsing

### PR DESCRIPTION
**Pull request by greglucas**
# Change Summary

## Overview
This follows the new file naming convention that the SPDF would like to follow. Changing some underscores to hyphens and adding an optional repointing field.

There is now a 4-digit repointing field that is `None` if it isn't present. Do we want to add `rp` or some other clarifier before that field to indicate what it is?

### CODE REVIEW

**Review by bourque — APPROVED** [review]
This looks good to me (if this is indeed the filename convention we can/should use ... it seems like there still might be some changes afoot from recent emails from Robert)

File: imap_data_access/__init__.py [hunk 1]
```diff
@@ -73,6 +73,6 @@
 VALID_FILE_EXTENSION = {"pkts", "cdf"}
```
Comment (tech3371) [inline imap_data_access/__init__.py:73]: can we update instruments name too?
```
VALID_INSTRUMENTS = {
    "codice",
    "glows",
    "hit",
    "hi",
    "idex",
    "lo",
    "mag",
    "swapi",
    "swe",
    "ultra",
}
```

Comment (tech3371) [reply imap_data_access/__init__.py:73]: Also, is there a command missing between "l1cb" and "l1d"?

File: imap_data_access/__init__.py [hunk 2]
```diff
@@ -73,6 +73,6 @@
...
 
 FILENAME_CONVENTION = (
-    "<mission>_<instrument>_<datalevel>_<descriptor>_"
-    "<startdate>_<enddate>_<version>.<extension>"
+    "<mission>_<instrument>_<datalevel>-<descriptor>_"
+    "<startdate>-<enddate>(-<repointing>?)_<version>.<extension>"
```
Comment (tech3371) [inline imap_data_access/__init__.py:77]:
```suggestion
    "<mission>_<instrument>_<datalevel>_<descriptor>_"
    "<startdate>(-<repointing>)_<version>.<extension>"
```

File: imap_data_access/file_validation.py [hunk 3]
```diff
@@ -26,21 +26,19 @@ def __init__(self, filename: str | Path):
...
             decided by each instrument. For L0, "raw" is used.
-        <startdate>: startdate is the earliest date in the data. Format - YYYYMMDD
-        <enddate>: Some instrument and some data level requires to store date range.
-            If there is no end date, then startdate will be used as enddate as well.
-            Format - YYYYMMDD.
-        <version>: This stores software version and data version. Version format is
-            vxx-xx.
+        <start_date>: startdate is the earliest date in the data, format: YYYYMMDD
+        <repointing?>: This is an optional field. It is used to indicate which
+            repointing the data is from, format: XXXXX
```
Comment (tech3371) [inline imap_data_access/file_validation.py:40]:
```suggestion
        <repointing>: This is an optional field. It is used to indicate which
            repointing the data is from. Repointing format is repoint followed by 5-digits: repointXXXXX
```

Comment (tech3371) [reply imap_data_access/file_validation.py:40]: We had a discussion on Slack about prefixing repoint to give it some meaning like how version number starts with `vxxx`. You mentioned about doing something like this for repointing, `repointXXXXX`. Do we still want to do this? That's why I suggested above change.

File: imap_data_access/file_validation.py [hunk 4]
```diff
@@ -254,11 +252,11 @@ def extract_filename_components(filename: str | Path) -> dict:
         pattern = (
             r"^(?P<mission>imap)_"
             r"(?P<instrument>[^_]+)_"
-            r"(?P<data_level>[^_]+)_"
+            r"(?P<data_level>[^-]+)_"
```
Comment (tech3371) [inline imap_data_access/file_validation.py:255]: should we undo this change based on recent discussion?

File: imap_data_access/file_validation.py [hunk 5]
```diff
@@ -254,11 +252,11 @@ def extract_filename_components(filename: str | Path) -> dict:
...
             r"(?P<descriptor>[^_]+)_"
-            r"(?P<start_date>\d{8})_"
-            r"(?P<end_date>\d{8})_"
-            r"(?P<version>v\d{2}-\d{2})"
+            r"(?P<start_date>\d{8})"
+            r"(-repoint(?P<repointing>\d{5}))?"  # Optional repointing field
```
Comment (tech3371) [inline imap_data_access/file_validation.py:258]: Oh cool. You are using prefix `repointXXXXX`

File: imap_data_access/__init__.py (hunk 1 above)
Comment (greglucas) [reply imap_data_access/__init__.py]: Good catch on the instruments, updated! I also updated the l1cb / l1d, which I think was a typo from before.

File: imap_data_access/file_validation.py (hunk 3 above)
Comment (greglucas) [reply imap_data_access/file_validation.py]: I added the `repoint` qualifier, but left off the extra text in the format section.

File: imap_data_access/file_validation.py (hunk 4 above)
Comment (greglucas) [reply imap_data_access/file_validation.py]: Yes, I think you're right, this should be underscore. I've updated it now!

File: imap_data_access/__init__.py (hunk 2 above)
Comment (greglucas) [reply imap_data_access/__init__.py]: Thanks for catching this, updated!

File: imap_data_access/cli.py [hunk 6]
```diff
@@ -242,14 +242,17 @@ def main():
     query_parser.add_argument(
```
Comment (tech3371) [inline imap_data_access/cli.py:242]: should we remove this now?

File: imap_data_access/cli.py [hunk 7]
```diff
@@ -242,14 +242,17 @@ def main():
...
         "--end-date", type=str, required=False, help="End date in YYYYMMDD format"
     )
+    query_parser.add_argument(
+        "--repointing", type=str, required=False, help="Repointing number"
```
Comment (tech3371) [inline imap_data_access/cli.py:246]:
```suggestion
        "--repointing", type=str, required=False, help="Repointing number. Eg. 'repointXXXXX'"
```

**Review by tech3371 — APPROVED** [review]
So far, it looks great to me. I will test it with infrastructure after you release it and open issue if I find any bugs. Thank you for doing this!
